File: apps/flutter-app/lib/common/model/group.dart
class Group {
  final String name;
  final int memberCount;
  final int transactionCount;

  Group({
    required this.name,
    required this.memberCount,
    required this.transactionCount,
  });

  factory Group.fromJson(Map<String, dynamic> json) {
    return Group(
      name: json['name'],
      memberCount: json['memberCount'],
      transactionCount: json['transactionCount'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'memberCount': memberCount,
      'transactionCount': transactionCount,
    };
  }
}

File: apps/go-api/models/group.go
package models

import "github.com/google/uuid"

type Group struct {
	Base
	Name         string        `json:"name" gorm:"type:varchar(100);required"`
	OwnerID      uuid.UUID     `json:"ownerId"`
	Members      []Member      `json:"members" gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
	Transactions []Transaction `json:"-" gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
	Debts        []Debt        `json:"-" gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
}

func (group *Group) IsMember(userID uuid.UUID) bool {
	for _, member := range group.Members {
		if member.UserID == userID {
			return true
		}
	}
	return false
}
